nima_core/memory_pruner.py

```
@@ -152,8 +152,25 @@ def get_candidates(
+    layer_placeholders = ", ".join(["?"] * len(layers))
+    sql = (
+        "SELECT id, text, timestamp, layer, "
+        "       COALESCE(fe_score, importance_hint, 0.5) AS importance, "
+        "       COALESCE(dismissal_count, 0) AS dismissal_count, "
+        "       COALESCE(themes, '') AS themes "
+        "FROM memory_nodes "
+        "WHERE COALESCE(is_ghost, 0) = 0 "
+        "  AND timestamp < ? "
+        f"  AND layer IN ({layer_placeholders}) "
+        "ORDER BY importance ASC "
+        "LIMIT 2000"
```

Ruff (0.15.7)

[error] Possible SQL injection vector through string-based query construction

(S608)